core/server/common/src/main/java/alluxio/master/journal/DelegatingGroupJournaled.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.journal;
+
+import alluxio.master.journal.checkpoint.CheckpointInputStream;
+import alluxio.proto.journal.Journal.JournalEntry;
+import alluxio.util.StreamUtils;
+
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Lists;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Convenience interface for classes which delegate journaling to a list of journaled components.
+ */
+public interface DelegatingGroupJournaled extends Journaled {
```

Comment:
Hrmmm, I feel like this is a generalized version of `DelegatingJournaled`. Ideally, this should probably replace DelegatingJournaled`, be merge into master, as an independent commit, so that change is isolated.

Reply:
I ported it to master with this PR: https://github.com/Alluxio/alluxio/pull/10151